test/models/orderbook.spec.ts

```diff
@@ -126,4 +126,58 @@ describe("Orderbook", () => {
       first_orderbook.add(second_orderbook);
     });
   });
+
+  it("Can compute the transitive closure of two orderbooks", () => {
+    const first_orderbook = new Orderbook("ETH", "DAI");
+    first_orderbook.addBid(new Offer(new Fraction(90, 1), 3));
+    first_orderbook.addBid(new Offer(new Fraction(95, 1), 2));
+    first_orderbook.addBid(new Offer(new Fraction(99, 1), 1));
+    first_orderbook.addAsk(new Offer(new Fraction(101, 1), 2));
+    first_orderbook.addAsk(new Offer(new Fraction(105, 1), 1));
+    first_orderbook.addAsk(new Offer(new Fraction(110, 1), 3));
+
+    const second_orderbook = new Orderbook("DAI", "USDC");
+    second_orderbook.addBid(new Offer(new Fraction(99, 100), 100));
+    second_orderbook.addBid(new Offer(new Fraction(9, 10), 200));
+    second_orderbook.addAsk(new Offer(new Fraction(101, 100), 100));
+    second_orderbook.addAsk(new Offer(new Fraction(105, 100), 200));
+
+    const closure = first_orderbook.transitiveClosure(second_orderbook);
+
+    assert.equal(closure.pair(), "ETH/USDC");
+    assert.equal(
+      JSON.stringify(closure.toJSON()),
+      JSON.stringify({
+        // The best bid eth_dai has enough liquidity on the best dai_usdc bid
+        // with 1 DAI remaining to be matched from the second best eth_dai bid (190 DAI).
+        // The remainder of the second best eth_dai bid (189 DAI) gets matched with the second
+        // best dai_usdc bid (200 DAI), leaving 11 DAI for the worst eth_dai bid (270 DAI).
+        // The remaining 259 DAI are unmatchable.
```

Comment:
Transitive orderbooks are complicated, but I think this comment does a good job at explaining why the expected values are what they are.